tests/libcxx/enc/CMakeLists.txt

```diff
@@ -92,31 +93,35 @@ foreach(testcase ${alltests})
             OR "${name}" MATCHES "sequences_array_at.pass" 
             OR "${name}" MATCHES "libcxx_containers_unord_next_pow2.pass" 
             OR "${name}" MATCHES "std_utilities_template.bitset_bitset.members_count.pass")
-            if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
-                continue()
-            endif()
+        if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR ENCLAVE_TESTS_CXX_COMPILER_ID MATCHES GNU)
+            continue()
+        endif()
     endif()
 
 # the following test fails when built with clang, see #830 -- Skipping this test in clang
     if ("${name}" MATCHES "array_sized_delete_array_calls_unsized_delete_array.pass")
-        if (CMAKE_CXX_COMPILER_ID MATCHES Clang)
+        if (CMAKE_CXX_COMPILER_ID MATCHES Clang OR ENCLAVE_TESTS_CXX_COMPILER_ID MATCHES Clang)
             continue()
         endif()
     endif()
 
 # the following test fails on GCC version 7, see  #1523 -- Skipping this test in GCC 7.0 -> 7.5
     if ("${name}" MATCHES "std_utilities_function.objects_comparisons_constexpr_init.pass")
-        if(CMAKE_CXX_COMPILER_ID MATCHES GNU
+        if((CMAKE_CXX_COMPILER_ID MATCHES GNU
                AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7
                AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.5)
+               OR (ENCLAVE_TESTS_CXX_COMPILER_ID MATCHES GNU
+               AND ENCLAVE_TESTS_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7
+               AND ENCLAVE_TESTS_CXX_COMPILER_VERSION VERSION_LESS 7.5))
             continue()
         endif()
     endif()
 
 # the following test fails on GCC 7 or newer, see #1559 -- Skip this test on GCC 7.0 and above in non debug builds
```

Comment:
Do you know here whether "non debug builds" includes CMAKE_BUILD_TYPE="RelWithDebebInfo"?